agents/appmotion_motor/appmotion_motors_driver.py

```diff
@@ -0,0 +1,702 @@
+                print("%s in receive mode." % (self.mot_id))
+            elif (msg != 'RS=R'):
+                print(
+                    "%s not in receive mode.  Resetting." %
+                    (self.mot_id))
+                print("Message was: ", msg)
+                self.kill_all_commands()
+                if (msg == 'RS=AR'):
+                    amsg = self.ser.writeread('AL\r')  # AL = Alarm Code
+                    print('is message is: ', amsg)
+                    print("Alarm was found. Resetting.")
+                    self.ser.write('AR\r')  # AR = Alarm Reset
+                    self.ser.flushInput()
+                else:
+                    print('Irregular message received.')
+                    sys.exit(1)
+
+        if m_res:
+            self.ser.write('EG8000\r')  # EG = Electronic Gearing
+            self.ser.write('SA\r')  # SA = Save Parameters
+            self.ser.flushInput()
+            sleep(0.1)
+            msg = self.ser.writeread('EG\r')
+            self.ser.flushInput()
+            if (len(msg) <= 4):    # Need at least MR=X + \r, which is 5 characters
+                print(
+                    "Couldn't get microstep resolution for %s.  Assuming 8,000." %
+                    (self.mot_id))
+            else:
+                print(msg)
+                ms_info = msg[3:]
+                self.s_p_rev = float(ms_info)
+        else:
+            msg = self.ser.writeread('EG\r')
+            self.ser.flushInput()
+            if (len(msg) <= 4):
+                print(
+                    "Couldn't get microstep resolution for %s. Disconnect and retry." %
+                    (self.mot_id))
+            else:
+                print(msg)
+                ms_info = msg[3:]
+                self.s_p_rev = float(ms_info)
+                ms_info = float(ms_info)
+
+        if (self.ser is not None) and (self.is_lin):
+            # DL1 = Define Limits for closed input (definition unclear in
+            # manual, however)
+            msg = self.ser.writeread('DL\r')
+            print(f"msg: {msg}")
+            if msg != 'DL=2':
+                print("Limits not defined as normally open. Resetting...")
+                self.ser.write('DL2\r')  # DL2 = Define Limits for open input
+                sleep(0.1)
+                self.ser.flushInput()
+            msg = self.ser.writeread('CC\r')  # CC = Change Current
+            print(msg)
+            current = float(msg[3:])
+            if current < 1.5:
+                print("Operating current insufficient. Resetting...")
+                self.ser.write('CC1.5\r')
+        else:
+            if self.ser is not None:
+                self.ser.write('JE\r')  # JE = Jog Enable
+
+    def is_moving(self, verbose=True):
+        """
+        Returns True if either motor is moving, False if both motors
+        are not moving. Also returns True if the motor provides an irregular
+        status message, such as any alarm keys.
```

Comment:
How does one distinguish between movement an alarm being tripped?

Reply:
Ah so, if the `verbose` flag is set to True, we print the error message and some helpful info. Thus:

-  if the motors are moving, we print (f'Motor {self.mot_id} is still moving.')
-  if the limit switch is hit, we print ('CCW limit switch hit unexpectedly.')

Based on error message that the motor controller sends